packages/stacking/src/index.ts

```diff
@@ -341,6 +519,144 @@ export class StackingClient {
+
+    if (poxAddress) {
+      const { hashMode, data } = decodeBtcAddress(poxAddress);
+      const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+      const hashbytes = bufferCV(data);
+      address = someCV(
+        tupleCV({
+          hashbytes,
+          version: hashModeBuffer,
+        })
+      );
+    }
+
+    const [contractAddress, contractName] = contract.split('.');
+    const network = this.network;
+    const txOptions: ContractCallOptions = {
+      contractAddress,
+      contractName,
+      functionName: 'delegate-stx',
+      functionArgs: [
+        uintCV(amountMicroStx.toString(10)),
+        standardPrincipalCV(delegateTo),
+        untilBurnBlockHeight ? uintCV(untilBurnBlockHeight) : noneCV(),
+        address ? address : noneCV(),
+      ],
+      validateWithAbi: true,
+      network,
+    };
+    return txOptions;
+  }
+
+  getDelegateStackOptions({
+    contract,
+    stacker,
+    amountMicroStx,
+    poxAddress,
+    burnBlockHeight,
+    cycles,
+    nonce,
+  }: {
+    contract: string;
+    stacker: string;
+    amountMicroStx: BN;
+    poxAddress: string;
+    burnBlockHeight: number;
+    cycles: number;
+    nonce?: BN;
+  }) {
+    const { hashMode, data } = decodeBtcAddress(poxAddress);
+    const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+    const hashbytes = bufferCV(data);
+    const address = tupleCV({
+      hashbytes,
+      version: hashModeBuffer,
+    });
+
+    const [contractAddress, contractName] = contract.split('.');
+    const network = this.network;
+    const txOptions: ContractCallOptions = {
+      contractAddress,
+      contractName,
+      functionName: 'delegate-stack-stx',
+      functionArgs: [
+        standardPrincipalCV(stacker),
+        uintCV(amountMicroStx.toString(10)),
+        address,
+        uintCV(burnBlockHeight),
+        uintCV(cycles),
+      ],
+      validateWithAbi: true,
+      network,
+    };
+
+    if (nonce) {
+      txOptions.nonce = nonce;
+    }
+
+    return txOptions;
+  }
+
+  getStackAggregationCommitOptions({
+    contract,
+    poxAddress,
+    rewardCycle,
+  }: {
+    contract: string;
+    poxAddress: string;
+    rewardCycle: number;
+  }) {
+    const { hashMode, data } = decodeBtcAddress(poxAddress);
+    const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+    const hashbytes = bufferCV(data);
+    const address = tupleCV({
+      hashbytes,
+      version: hashModeBuffer,
+    });
+
+    const [contractAddress, contractName] = contract.split('.');
+    const network = this.network;
+    const txOptions: ContractCallOptions = {
+      contractAddress,
+      contractName,
+      functionName: 'stack-aggregation-commit',
+      functionArgs: [address, uintCV(rewardCycle)],
+      validateWithAbi: true,
+      network,
+    };
+    return txOptions;
+  }
+
+  getRevokeDelegateStxOptions(contract: string) {
+    const [contractAddress, contractName] = contract.split('.');
```

Comment:
Maybe we should add some error handling here incase an erroneous string is added?

Reply:
something like this (before passing the contract to the `getXOptions` methods:

```
const contract : string = poxInfo.contract_id;

    if(contract.length === 0 || contract.split('.').length !== 2) {
      throw new Error('Fetching Stacking contract ID is malformed');
    }
```